src/core/ChannelHandler.js

```diff
@@ -0,0 +1,103 @@
+import Observable from 'zen-observable';
+
+const events = [
+  'phx_reply',
+  'phx_error',
+  'status_changed',
+  'stopped',
+  'error',
+  'failed',
+  'starting',
+  'started',
+  'deploying',
+  'deployed',
+  'deployment_failed',
+  'ready',
+  'terminated',
+  'snapshot_taken',
+  'snapshot_reverted'
+];
+
+export default class ChannelHandler {
+  constructor(name, socket, globalLogger) {
+    this._name = name;
+    this._channel = socket.channel(this._name);
+    this._stream = this._buildChannelStream(events);
+    this._log = globalLogger.extend(`${this._name}`);
+    this._logger = this._stream.subscribe(value =>
+      this._log(JSON.stringify(value, null, 4))
+    );
+    this.init();
+  }
+
+  _buildChannelStream(eventsList) {
+    return new Observable(stream => {
+      const setChannelEvent = event => {
```

Comment:
So `buildChannelStream` builds a observable stream, watching for one `socket.io channel event listener` per event type available.

1 - why not just "watch for all events" on one event listener instead of having dozens of individual event listeners in memory?
2 - this closure here feels unnecessary, you can just replace it with what you wrote below https://github.com/makerdao/testchain-client/pull/19/files?utf8=%E2%9C%93&diff=unified#diff-167640be9e84a7ea5778483c08a09be5R59

Reply:
> So buildChannelStream builds a observable stream, watching for one socket.io channel event listener per event type available.

`phoenix.js` not socket.io, but yes. There's no wildcard, listen for all events on the channel level. We could listen directly on the socket for all events on all chains but would require parsing of each chain and may run into issues with more scale possibly?

Reply:
I think a more efficient way we could work around this is to use the Observable that exists on the socket which enables us to subscribe on all data. When we wish to create a `ChannelHandler` when instantiating a new chain, we can pass the filtered observable as a constructor arg